### PR TITLE
Implement isExported for symbols in macros

### PR DESCRIPTION
**Pull request by nc-x**
Closes https://github.com/nim-lang/Nim/issues/11878

### CODE REVIEW

**Comment by nc-x** [comment]
Unrelated test failure.

**Comment by Araq** [comment]
Please use the VM's callback mechanism to implement this, not yet another magic+opcode.

**Comment by nc-x** [comment]
@Araq 
Done. And it turned out to be much simpler too. :smile:

File: compiler/vmops.nim [hunk 1]
```diff
@@ -148,3 +148,8 @@ proc registerAdditionalOps*(c: PCtx) =
     let n = getNode(a, 0)
     if n.kind != nkSym: raise newException(ValueError, "node is not a symbol")
     setResult(a, $symBodyDigest(c.graph, n.sym))
+
+  registerCallback c, "stdlib.macros.isExported", proc(a: VmArgs) {.nimcall.} =
+    let n = getNode(a, 0)
+    if n.kind != nkSym: raise newException(ValueError, "node is not a symbol")
```
Comment (Araq) [inline compiler/vmops.nim]: The VM doesn't handle exceptions... Simply return ``false`` then?

Comment (krux02) [reply compiler/vmops.nim]: I would use ``template stackTrace(c: PCtx, tos: PStackFrame, pc: int, msg: string)`` here.

Comment (Araq) [reply compiler/vmops.nim]: Or that, yes.

Comment (nc-x) [reply compiler/vmops.nim]: Done.

**Pull request merged**